From kph033@ikp073.ikp.kfa-juelich.de Wed Apr 25 06:10:06 2007
From: "t.sefzick" <t.sefzick@fz-juelich.de>
Date: Wed, 25 Apr 2007 15:05:22 +0200
Subject: USB: flow control fix for pl2303
To: greg@kroah.com
Cc: t.sefzick@fz-juelich.de
Message-ID: <20070425130522.GA12888@ikp073.ikp.kfa-juelich.de>
Content-Disposition: inline

From: "t.sefzick" <t.sefzick@fz-juelich.de>

in order to be able to switch back to 'flow-control none'
after having activated 'flow-control rts/cts', I made
a small change to 'pl2303.c'.


Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/pl2303.c |    7 +++++++
 1 file changed, 7 insertions(+)

--- a/drivers/usb/serial/pl2303.c
+++ b/drivers/usb/serial/pl2303.c
@@ -617,6 +617,13 @@ static void pl2303_set_termios(struct us
 				    VENDOR_WRITE_REQUEST_TYPE,
 				    0x0, index, NULL, 0, 100);
 		dbg("0x40:0x1:0x0:0x%x  %d", index, i);
+	} else {
+		i = usb_control_msg(serial->dev,
+				    usb_sndctrlpipe(serial->dev, 0),
+				    VENDOR_WRITE_REQUEST,
+				    VENDOR_WRITE_REQUEST_TYPE,
+				    0x0, 0x0, NULL, 0, 100);
+		dbg ("0x40:0x1:0x0:0x0  %d", i);
 	}
 
 	kfree(buf);
